library/src/main/java/software/amazon/location/auth/LocationCredentialsProvider.kt

```diff
@@ -82,6 +82,24 @@ class LocationCredentialsProvider {
         }
     }
 
+    suspend fun generateCredentials(identityPoolId: String, region: Regions): Boolean {
+        val cognitoCredentialsProvider = CognitoCredentialsProvider(region.getName())
+        try {
+            val identityId = cognitoCredentialsProvider.getIdentityId(identityPoolId)
+            if (identityId.isNotEmpty()) {
+                val credentials = cognitoCredentialsProvider.getCredentials(identityId)
+                awsKeyValueStore.put("accessKeyId", credentials.credentials.accessKeyId)
+                awsKeyValueStore.put("secretKey", credentials.credentials.secretKey)
+                awsKeyValueStore.put("sessionToken", credentials.credentials.sessionToken)
+                awsKeyValueStore.put("expiration", credentials.credentials.expiration.toString())
+                return true
+            }
+        } catch (e: Exception) {
+           e.printStackTrace()
+        }
+        return false
```

Comment:
if credentials could not be generated, the auth call should throw an exception, not fail silently

Reply:
Done